django/contrib/syndication/views.py

```diff
@@ -86,10 +86,17 @@ def _get_dynamic_attr(self, attname, obj, default=None):
                 code = attr.__code__
             except AttributeError:
                 code = attr.__call__.__code__
-            if code.co_argcount == 2:  # one argument is 'self'
-                return attr(obj)
-            else:
-                return attr()
+            if not code.co_argcount:  # attr was decorated
+                if hasattr(attr, "__wrapped__"):
+                    code = attr.__wrapped__.__code__
```

Comment:
Can we use `unwrap()` instead? Also this doesn't handle wrapped callable objects e.g.:
```python
class MyCallable:
    @custom_decorator
    def __call__(self):
        return 1
```
Maybe we should unwrap in advance :thinking: 
```python
        if callable(attr):
            try:
                code = unwrap(attr).__code__
            except AttributeError:
                code = unwrap(attr.__call__).__code__
            if not code.co_argcount: 
                raise ImproperlyConfigured(...)
```